wandb/sdk/launch/agent/agent.py

```diff
@@ -319,9 +315,11 @@ def finish_thread_id(
             except CommError:
                 pass
             if run_info is None:
-                _msg = "The submitted run was not successfully started"
                 fnames = None
-
+                if job_and_run_status.completed_status == "finished":
+                    _msg = "The submitted job exited successfully but failed to call wandb.init"
+                else:
+                    _msg = "The submitted run was not successfuly started"
```

Comment:
This message still seems incomplete, I think we can say:
"The submitted job failed, and did not call initialize a run"